modules/spring-config/src/main/java/org/mule/config/spring/XmlMetadataFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.config.spring;
+
+import org.mule.config.spring.parsers.XmlMetadataAnnotations;
+
+import org.xml.sax.Locator;
+
+/**
+ * Factory object to get a fresh instance of a {@link XmlMetadataAnnotations}.
+ * 
+ * This interface is an extension point to other Mule projects. Please be careful if you are going to make changes here.
+ */
+public interface XmlMetadataFactory
```

Comment:
Minor, but better IMO to use:
i) XmlMetadataAnnotationsFactory
ii) create() rather than createWith(), to be more consistent with other factories in Mule which don't attempt to be fluent.
WDYT?